WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -341,7 +341,6 @@ def execute(self):
       result = self._submitJob(jobID, params, ceDict, optimizerParams, proxyChain,
                                processors, wholeNode, maxNumberOfProcessors, mpTag)
       if not result['OK']:
-        self.__report(jobID, 'Failed', result['Message'])
```

Comment:
Why do you drop this ? If we are here, the job is not successful but it is not rescheduled neither. So, either it should be Failed or we try to reschedule it. I think we have to reschedule it since it is unlikely the problem of the user payload.

Reply:
The `JobWrapper` might set the status to `Done`, which might then be overridden here. We are seeing these cases (look at the last 2 lines):
```
[localhost] ~ $ dirac-wms-job-logging-info 349824390
Source               Status     MinorStatus                                         ApplicationStatus    DateTime
==============================================================================================================================
JobManager           Received   Job accepted                                        Unknown              2020-03-15 08:49:06
JobPath              Checking   JobSanity                                           Unknown              2020-03-15 08:49:06
JobSanity            Checking   AncestorFiles                                       Unknown              2020-03-15 08:49:06
AncestorFiles        Checking   AncestorFiles                                       Unknown              2020-03-15 08:49:28
InputData            Checking   JobScheduling                                       Unknown              2020-03-15 08:50:43
JobScheduling        Waiting    Pilot Agent Submission                              Unknown              2020-03-15 08:50:43
Matcher              Matched    Assigned                                            Unknown              2020-03-15 08:53:08
JobAgent@LCG.RAL.uk  Matched    Job Received by Agent                               Unknown              2020-03-15 08:53:08
JobAgent@LCG.RAL.uk  Matched    Submitted To CE                                     Unknown              2020-03-15 08:53:09
JobWrapper           Running    Job Initialization                                  Unknown              2020-03-15 08:53:12
JobWrapper           Running    Downloading InputSandbox                            Unknown              2020-03-15 08:53:13
JobWrapper           Running    Input Data Resolution                               Unknown              2020-03-15 08:53:14
JobWrapper           Running    Application                                         Unknown              2020-03-15 08:57:34
Job_349824390        Running    Application                                         Executing DaVinci_1  2020-03-15 08:57:38
Job_349824390        Running    Application                                         DaVinci step 1       2020-03-15 08:57:39
JobWrapper           Completed  Application Finished Successfully                   DaVinci step 1       2020-03-15 19:30:03
JobWrapper           Completed  Uploading Output Sandbox                            DaVinci step 1       2020-03-15 19:30:07
JobWrapper           Completed  Output Sandbox Uploaded                             DaVinci step 1       2020-03-15 19:30:16
JobWrapper           Done       Execution Complete                                  DaVinci step 1       2020-03-15 19:30:17
JobAgent@LCG.RAL.uk  Failed     InProcess CE Error: InProcess Job Execution Failed  DaVinci step 1       2020-03-15 19:31:16
```

Reply:
I see. The job process segfaults while the final cleanup of the working directory. Indeed, in this case the job itself is OK.